src/App.js

```diff
@@ -457,6 +446,12 @@ export default class App extends React.Component<{}, AppState> {
         });
     }
 
+    handleChangeProgramSpeed = (e: SyntheticInputEvent<HTMLInputElement>) => {
+        const speedLookUp = [2000, 1500, 1000, 500, 250];
+        const newMovementDelay = speedLookUp[parseInt(e.target.value) - 1];
+        this.interpreter.timeDelayMs = newMovementDelay;
```

Comment:
Add a method to Interpreter `setStepTime()`, rather than setting the property directly.